decidim-surveys/db/migrate/20240925124312_add_settings_to_decidim_surveys_surveys.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AddSettingsToDecidimSurveysSurveys < ActiveRecord::Migration[7.0]
+  def change
+    add_column :decidim_surveys_surveys, :starts_at, :datetime
+    add_column :decidim_surveys_surveys, :ends_at, :datetime
+    add_column :decidim_surveys_surveys, :announcement, :jsonb
+    add_column :decidim_surveys_surveys, :allow_answers, :boolean
+    add_column :decidim_surveys_surveys, :allow_unregistered, :boolean
+    add_column :decidim_surveys_surveys, :clean_after_publish, :boolean
+    add_column :decidim_surveys_surveys, :published_at, :datetime, index: true
+  end
```

Comment:
```suggestion
  class Survey < ApplicationRecord
    self.table_name = :decidim_surveys_surveys
  end

  def up
    add_column :decidim_surveys_surveys, :starts_at, :datetime
    add_column :decidim_surveys_surveys, :ends_at, :datetime
    add_column :decidim_surveys_surveys, :announcement, :jsonb
    add_column :decidim_surveys_surveys, :allow_answers, :boolean
    add_column :decidim_surveys_surveys, :allow_unregistered, :boolean
    add_column :decidim_surveys_surveys, :clean_after_publish, :boolean
    add_column :decidim_surveys_surveys, :published_at, :datetime, index: true
    
    Survey.update(published_at: Time.current)
  end

  def down
    remove_column :decidim_surveys_surveys, :starts_at
    remove_column :decidim_surveys_surveys, :ends_at
    remove_column :decidim_surveys_surveys, :announcement
    remove_column :decidim_surveys_surveys, :allow_answers
    remove_column :decidim_surveys_surveys, :allow_unregistered
    remove_column :decidim_surveys_surveys, :clean_after_publish
    remove_column :decidim_surveys_surveys, :published_at
  end
```

We need to add a default value to existing surveys, otherwise all the existing surveys will be unpublished!

Reply:
Done

Reply:
Actually I see that we should not use the "Time.current" value, but instead the `created_at` time of the survey, something like:

```ruby
Survey.find_each do |survey|
  survey.update(published_at: survey.created_at)
end
```

Reply:
Done